### PR TITLE
fix dnspolicy initializer

### PR DESCRIPTION
**Pull request by handlerww**
### What problem does this PR solve?

fix https://github.com/pingcap/tidb-operator/pull/4420, I missed podspec initializer.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Review by july2993 — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

File: pkg/apis/pingcap/v1alpha1/tidbcluster_component.go [hunk 1]
```diff
@@ -428,6 +429,7 @@ func buildDMClusterComponentAccessor(c Component, dc *DMCluster, componentSpec *
 		clusterAnnotations:        spec.Annotations,
 		tolerations:               spec.Tolerations,
 		dnsConfig:                 spec.DNSConfig,
+		dnsPolicy:                 spec.DNSPolicy,
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/tidbcluster_component.go:432]: Please also update `pkg/manager/utils/statefulset_builder.go:	b.prototype.Spec.DNSPolicy = corev1.DNSClusterFirstWithHostNet`.

Comment (handlerww) [reply pkg/apis/pingcap/v1alpha1/tidbcluster_component.go:432]: fixed

**Pull request merged**